h2o-core/src/main/java/water/api/DatasetServlet.java

```diff
@@ -31,7 +31,11 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) {
           .setHexString(Boolean.parseBoolean(hex_string))
           .setEscapeQuotes(Boolean.parseBoolean(escape_quotes_string));
       InputStream is = dataset.toCSV(parms);
-      response.setContentType("application/octet-stream");
+      if (uri.equals("/3/DownloadDataset")) {
```

Comment:
`request.getServletPath()` is more appropriate than `uri` if you're going to use `equals`

Reply:
another way would be to subclass `DownloadDataset` with `DownloadDatasetBin` and have a final content type defined for each...